### ISSUE TITLE
Provinces page enhancements

### ISSUE THREAD

**Issue by zainfathoni** [kawalcovid19/wargabantuwarga.com#110]
## Overview

Since we have finished our basic functionalities for the [Provinces page](https://www.wargabantuwarga.com/provinces), we need to enhance them into our users' best possible experience.

## Current Tasks

- [ ] Use a proper client-side search engine library to power the search functionality in our pages
- [ ] ...

**Comment by baraeb92** [comment]
Wishing the following enhancement as well: 

1. Sort by alpabhetical order
2. Search or Filter by Category
3. Button for direction and call directly from the list.

**Comment by baraeb92** [comment]
Something like this![20210718_014641.jpg](https://user-images.githubusercontent.com/23743497/126046806-1f32fa28-af71-46d9-b519-c16dce7fba3b.jpg)

**Comment by aldiharyo** [comment]
Setelah filter by kota, next stepnya filter based on needs seperti di https://indiacovidresources.in itu. Saya lihat sudah ada tiketnya by category itu yang seperti ini bukan?

![by category](https://user-images.githubusercontent.com/87590228/126054003-86c09f93-57e7-4977-8a80-46ddb7b62241.jpg)

**Comment by zainfathoni** [comment]
No, it's not. What I meant in #112 is a dropdown filter like this:

<img width="268" alt="Screenshot 2021-07-18 at 10 49 22" src="https://user-images.githubusercontent.com/6315466/126054078-fc154d0e-191a-4116-af0e-47f1144cc725.png">

If we want to have a different experience like that, I think it is worth a separate issue due to its complexity. 😁 
And I think it's better to wait for the final UI/UX design from Mas @baraeb92 & Mas Momo.

**Comment by zainfathoni** [comment]
With #374 coming, I'm wondering whether we still need #125 & #197 implemented. 🤔 

Any opinions, @kawalcovid19/wbw?

**Comment by ekafyi** [comment]
🤔 I have no real-world experience with this tbh, but the new typesense will enable us to fetch in paginated batch (eg. 30 first items, next 30 items, etc), is that correct? In that case IMO we won't need virtualized lists.

**Comment by zainfathoni** [comment]
> 🤔 I have no real-world experience with this tbh, but the new Typesense will enable us to fetch in paginated batch (eg. 30 first items, next 30 items, etc), is that correct? In that case, IMO we won't need virtualized lists.

What do you think, Mas @famasya?
Typesense provides proper pagination, doesn't it?

**Comment by famasya** [comment]
> > 🤔 I have no real-world experience with this tbh, but the new Typesense will enable us to fetch in paginated batch (eg. 30 first items, next 30 items, etc), is that correct? In that case, IMO we won't need virtualized lists.
> 
> What do you think, Mas @famasya?
> Typesense provides proper pagination, doesn't it?

Yes, it provides decent pagination as described [here](https://typesense.org/docs/0.21.0/api/documents.html#arguments). FYI, it also provides federated search (search over multiple indexes at once) to reduce round-trip latency.
